bb-pilcom/bb-pil-backend/templates/circuit_builder.cpp.hbs

```diff
@@ -10,33 +10,37 @@
 #include "barretenberg/relations/generic_permutation/generic_permutation_relation.hpp"
 #include "barretenberg/relations/generic_lookup/generic_lookup_relation.hpp"
 #include "barretenberg/honk/proof_system/logderivative_library.hpp"
+#include "barretenberg/numeric/bitop/get_msb.hpp"
 #include "barretenberg/vm/stats.hpp"
 
 namespace bb {
 
 {{name}}CircuitBuilder::ProverPolynomials {{name}}CircuitBuilder::compute_polynomials() const {
-    const auto num_rows = get_circuit_subgroup_size();
+    const size_t circuit_subgroup_size = get_circuit_subgroup_size();
+    // FIXME: Either some algo or the Polynomial class seems to require this to be a power of 2.
+    const size_t num_rows = numeric::round_up_power_2(get_num_gates());
     ProverPolynomials polys;
 
     // Allocate mem for each column
     AVM_TRACK_TIME("circuit_builder/init_polys_to_be_shifted", ({
                        for (auto& poly : polys.get_to_be_shifted()) {
                            poly = Polynomial{ /*memory size*/ num_rows - 1,
-                                              /*largest possible index*/ num_rows,
+                                              /*largest possible index*/ circuit_subgroup_size,
```

Comment:
so to clarify, this PR allows you to set some max trace size circuit_subgroup_size (at 2^21) but the polynomial time/memory is restricted to num_rows. What hasn't been done yet is further optimization of things like selectors which are all 0, etc...